src/CUITe/Controls/ControlBase.cs

```diff
@@ -124,11 +134,38 @@ public void WaitForControlReady()
         }
 
         /// <summary>
+        /// Waits for the control to be ready.
+        /// </summary>
+        protected void WaitForControlReadyIfNecessary()
+        {
+            if (isControlReadinessAwaitedByDefault)
+            {
+                sourceControl.WaitForControlReady();
+            }
+        }
+
+        /// <summary>
+        /// Waits for the control to appear in the user interface.
+        /// </summary>
+        public void WaitForControlExist()
+        {
+            sourceControl.WaitForControlExist();
+        }
+
+        /// <summary>
+        /// Waits for the control to appear in the user interface, or for the specified timeout to expire.
+        /// </summary>
+        public void WaitForControlExist(int millisecondsTimeout)
```

Comment:
1. XML documentation is missing for the millisecondsTimeout parameter.
```
/// <param name="millisecondsTimeout">The number of milliseconds before time-out.</param>
```

2. Return the value returned from `sourceControl.WaitForControlExist(millisecondsTimeout);`
```
/// <returns><code>true</code> if this control exists before the time-out; otherwise, <code>false</code>.</returns>
public int WaitForControlExist(int millisecondsTimeout)
{
    return sourceControl.WaitForControlExist(millisecondsTimeout);
}
```